code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -576,3 +576,54 @@
 				set_light(1, 2, "#5dca31")
 
 #undef RAD_COST
+
+
+/////////////////////////////////////////////////////// 
+///////////////// HEADCRAB ZOMBIES //////////////////// 
+///////////////////////////////////////////////////////
+
+/mob/living/simple_animal/hostile/necro/zombie/headcrab //Not very useful
+	icon_state = "zombie_turned" //Looks almost not unlike just a naked guy to potentially catch others off guard
+	icon_living = "zombie_turned"
+	icon_dead = "zombie_turned"
+	desc = "A reanimated corpse that looks like it has seen better days. This one still appears quite human."
+	maxHealth = 75
+	health = 75
+	can_evolve = FALSE
+	canRegenerate = 1
+	harm_intent_damage = 20
+	melee_damage_lower = 15
+	melee_damage_upper = 2
+	var/mob/living/carbon/human/host //Whoever the zombie was previously, kept in a reference to potentially bring back
+
+/mob/living/simple_animal/hostile/necro/zombie/headcrab/New()
+	..()
+
+/mob/living/simple_animal/hostile/necro/zombie/headcrab/Destroy()
+	if(host)
+		qdel(host)
+		host = null
+	..()
+
+/mob/living/simple_animal/hostile/necro/zombie/headcrab/death(var/gibbed = FALSE)
+	if(host && mind)
+		host.loc = get_turf(src)
+		mind.transfer_to(host)
+		var/mob/dead/observer/ghost = get_ghost_from_mind(mind)
+		if(ghost && ghost.can_reenter_corpse)
+			key = mind.key
+		if(prob(33))	//33% chance to blow up their fucking head
+			var/datum/organ/external/head/head_organ = host.get_organ(LIMB_HEAD)
+			head_organ.explode()
+		else
+			visible_message("<span class='danger'>The headcrab releases it's grasp from [src]!</span>")
+		host = null
```

Comment:
You might want to qdel the host before setting this to null

Reply:
No you don't, setting host to null here is unnecessary to begin with. You only need to clear refs to the datum that you're deleting, not the ones inside of the datum you're deleting.